### PR TITLE
update mobile UI to polymer 1.0

### LINKED ISSUE #444: upgrade our mobile UI to polymer 1.0?
https://elements.polymer-project.org/browse?package=iron-elements
https://elements.polymer-project.org/browse?package=paper-elements

bower dependencies should be updated to:

```
"iron-elements": "PolymerElements/iron-elements#^1.0.0",
"paper-elements": "PolymerElements/paper-elements#^1.0.0"
```

### PR DESCRIPTION
**Pull request by kasperpeulen**
fix #444 

@devoncarew @lukechurch 

I think I lost a couple of animation, riddles etc, but the basic functionality should work the same as in our 0.5 version. There is a lot changed, so there are probably a couple of new bugs that I haven't catched yet. 

I have changed it so that the node structure is build in `mobile.html`, in that way I could easier debug, understand what was going on etc. If there is some reason to prefer having the nodes build in `mobile.dart`, I could change this back of course.

Some things are probably not done in the cleanest way possible. I probably will look at this code a couple of more times, trying to look for nicer solutions, but at least it all works :)

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -17,33 +17,29 @@
...
-    "core-item": "polymer/core-item#^0.5",
-    "core-menu": "polymer/core-menu#^0.5",
-    "paper-menu-button": "polymer/paper-menu-button#^0.5",
-    "core-splitter": "polymer/core-splitter#^0.5",
-    "webcomponentsjs": "polymer/webcomponentsjs#^0.5"
-  },
-  "resolutions": {
-    "webcomponentsjs": "~0.5.5"
+    "polymer": "Polymer/polymer#~1.0.2",
+    "iron-flex-layout": "PolymerElements/iron-flex-layout#^1.0.0",
```
Comment (devoncarew) [inline bower.json]: It'd may be possible to replace most of these imports with:
```
"iron-elements": "PolymerElements/iron-elements#^1.0.0",
"paper-elements": "PolymerElements/paper-elements#^1.0.0"
```
but I haven't tried it -

File: lib/polymer/iron.dart [hunk 2]
```diff
@@ -10,82 +10,26 @@ import 'dart:js';
...
-class CoreMenu extends CoreSelector {
-  CoreMenu() : super('core-menu');
-  CoreMenu.from(HtmlElement element) : super.from(element);
-}
-
-class CoreOverlay extends CoreElement {
-  CoreOverlay([String tag]) : super(tag == null ? 'core-overlay' : tag);
-  CoreOverlay.from(HtmlElement element) : super.from(element);
+class IronOverlayBehavior extends PolymerElement {
+  IronOverlayBehavior([String tag]) : super(tag == null ? 'core-overlay' : tag);
```
Comment (devoncarew) [inline lib/polymer/iron.dart]: Should this be `iron-overlay` instead of `core-overlay`?

File: lib/polymer/iron.dart [hunk 3]
```diff
@@ -105,46 +49,40 @@ class CoreOverlay extends CoreElement {
...
-
-  /**
-   * Fired when a selection is activated.
-   */
-  Stream get onCoreActivate => listen('core-activate');
+  Stream get ironActivate => listen('iron-activate');
 }
 
-class CoreSplitter extends CoreElement {
+class CoreSplitter extends PolymerElement {
```
Comment (devoncarew) [inline lib/polymer/iron.dart]: Is this still the latest? I can't find a splitter in https://elements.polymer-project.org/browse.

File: lib/polymer/iron.dart [hunk 4]
```diff
@@ -249,18 +187,36 @@ class CoreElement extends WebElement {
...
 
 class Transition {
-  static void coreTransitionCenter(CoreElement element) =>
+  static void coreTransitionCenter(PolymerElement element) =>
       element.setAttribute('transition', 'core-transition-center');
 }
 
 class Transitions {
-  static void slideFromRight(CoreElement element) =>
+  static void slideFromRight(PolymerElement element) =>
```
Comment (devoncarew) [inline lib/polymer/iron.dart]: I'm wondering if the old transitions map to the new `neon-animation`s (https://elements.polymer-project.org/elements/neon-animation?active=slide-from-right-animation).

**Comment by devoncarew** [comment]
Awesome! Very comprehensive overhaul!

Can you change this to be a PR into a `polymer` branch (instead of `master`)? I'd like to be able to iterate on this before we pull it into the main branch.

**Comment by devoncarew** [comment]
Also, can you attach a screenshot of the UI (for posterity). And I see one analysis warning here:
```
[info] The value of the field '_polymer' is not used at lib/polymer/polymer.dart, line 15.
```

File: lib/polymer/iron.dart (hunk 2 above)
Comment (kasperpeulen) [inline lib/polymer/iron.dart]: I wasn't completely sure what to do with this. Iron-overlay doesn't exist I think, you know have `behaviors` with a couple of methods,property attached to it, that you can implement, but not construct. You can't extend elements anymore in polymer 1.0. 

So there is now an iron-overlay tag, but you have `Polymer.IronOverlayBehavior`:

https://elements.polymer-project.org/elements/iron-overlay-behavior?active=Polymer.IronOverlayBehavior

I'm not really sure how to `translate` behaviors to dart.

File: lib/polymer/iron.dart (hunk 3 above)
Comment (kasperpeulen) [inline lib/polymer/iron.dart]: I haven't updated this one indeed. I think we can maybe delete this one, as we don't use a splitter in the mobile version I think.

File: lib/polymer/iron.dart (hunk 4 above)
Comment (kasperpeulen) [inline lib/polymer/iron.dart]: Yeah, I still need to update those transitions. I have looked at those neon animation. I use it for the dropdown animation (in the right top). Will try to get the slidefromRight transition working. I also need to get the riddles in the left menu back.

**Comment by kasperpeulen** [comment]
>Can you change this to be a PR into a polymer branch (instead of master)? I'd like to be able to iterate on this before we pull it into the main branch

I don't know what you mean with this. It is now in my `kasperpeulen:polymer1.0` branch, do you want me to merge it to some `dart-lang:polymer` branch ? If so, I think I can't create such a branch for `dart-lang`.

>Also, can you attach a screenshot of the UI (for posterity). 



see also kasperpeulen.github.io/dart-pad/mobile.html

> [info] The value of the field '_polymer' is not used at lib/polymer/polymer.dart, line 15.

I don't use this field anymore, but I thought that it could be useful in the future. Maybe comment it out ?

**Comment by devoncarew** [comment]
> I don't use this field anymore, but I thought that it could be useful in the future. Maybe comment it out ?

sgtm

**Comment by devoncarew** [comment]
OK, I pushed your current branch to https://github.com/dart-lang/dart-pad/tree/polymer. Let's work off that one, and plan to merge it to master when we're happy with it. Sound good?

**Comment by kasperpeulen** [comment]
I guess I can close this one than @devoncarew

**Pull request closed without merging**